src/optionstr.c

```diff
@@ -1552,56 +1552,64 @@ did_set_commentstring(optset_T *args)
 #endif
 
 /*
- * The 'complete' option is changed.
+ * Check if value for 'complete' is valid when 'complete' option is changed.
  */
     char *
 did_set_complete(optset_T *args)
 {
     char_u	**varp = (char_u **)args->os_varp;
-    char_u	*s;
+    char_u	*p;
```

Comment:
```suggestion
    char_u	*p = NULL;
```
always declare a variable with a default value. also in some other functions.